app/controllers/katello/api/v2/api_controller.rb

```diff
@@ -192,6 +192,14 @@ def find_default_organization_and_or_environment
       end
     end
 
+    def respond_for_show(options = {})
+      if (options.try(:[], :resource).try(:organization_id) && params[:organization_id] &&
+         (options[:resource].organization_id.to_s != params[:organization_id].to_s))
+        fail HttpErrors::NotFound, _("Resource does not belong to the specified organization")
+      end
+      super
+    end
+
```

Comment:
So using this we can check for resource organization for show methods if organization id is passed to the API. For cases where organization id is not passed to the API, the UI will show the warning as earlier..